C++/include/algorithm/dynamic_programming/longest_decreasing_subsequence.hpp

```diff
@@ -0,0 +1,69 @@
+#ifndef LONGEST_DECREASING_SUBSEQUENCE
+#define LONGEST_DECREASING_SUBSEQUENCE
+
+/*
+Longest Decreasing Subsequence
+---------------------
+This algorithm finds the longest decreasing subsequence in a sequence of numbers
+
+Time complexity
+----------------
+O(n log(n)) where n is the number of elements in the initial list
+
+Space complexity
+---------------
+O(n)
+
+*/
+
+#include <iostream>
+#include <vector>
+/*
+ *	Function finds the longest_decreasing_subsequence for a vector input
+ */
+int longest_decreasing_subsequence(std::vector<int>& input,  const bool to_show_state = true) {
```

Comment:
If you include

```
using std::vector;
using std::cout;
using std::endl;
```
at the top, then you can just write `vector` instead of `std::vector` (for example). This will make your code a lot cleaner.